openassessment/xblock/static/js/src/lms/oa_datefactory.js

```diff
@@ -0,0 +1,90 @@
+
+/**
+ *
+ * A helper function to utilize DateUtils.
+ **/
+
+OpenAssessment.DateTimeFactory = function(element) {
+    this.element = element;
+};
+
+OpenAssessment.DateTimeFactory.prototype = {
+
+    report: function() {
```

Comment:
Is there a better name for `report`?  Since this isn't really creating a report.

How about renaming `report` to `apply` (or `apply_to_doc`) and `generate` to `apply_to_element`?   Since the former affects the entire document, while the latter affects a single element.